### PR TITLE
Automatically scale noise level in GaussianProcessLearner

### PR DESCRIPTION
**Pull request by zakv**
The machine learning optimizers normalize the costs before fitting such that they have zero mean and standard deviation one, and they scale other related parameters as well. Generally these conversions between real units and normalized quantities happen behind the scenes so that the user only has to think in real units. That's particularly important since the scaling parameters needed to normalize the costs aren't known ahead of time and change throughout the optimization.

Previously the `noise_level` argument for the `GaussianProcessLearner` was passed directly to scikit learn's `WhiteKernel` without being scaled. This made it difficult for users to specify a reasonable value as they'd have to estimate what they think the scaling parameters would be. This PR proposes to change the behavior so that the user specifies the `noise_level` in real units, then lets `GaussianProcessLearner` perform the necessary scaling behind the scenes. This also has the benefit that the figure generated by `GaussianProcessVisualizer.plot_noise_level_vs_run()` displays the results in real units as well.

For now I've left it such that `noise_level` quantifies the uncertainty as a variance, not a standard deviation, in the cost. That's likely not exactly what the user would expect, especially given that the uncertainty passed with the cost during the optimization should be a standard deviation. However it is consistent with the previous behavior and the meaning of the identically named `noise_level` argument of [`WhiteKernel`](https://scikit-learn.org/stable/modules/generated/sklearn.gaussian_process.kernels.WhiteKernel.html#). To try to ease this confusion I've updated the docs with a brief explanation. I'd be happy to change the behavior if desired, but I thought I'd err on the side of keeping things the same.

Changes proposed in this pull request:

- Make `GaussianProcessLearner` scale user-provided values for `noise_level`.
- Update docs.

@qctrl/support @charmasaur

### CODE REVIEW

File: mloop/learners.py [hunk 1]
```diff
@@ -1299,6 +1312,15 @@ def fit_gaussian_process(self):
         self.scaled_costs = self.cost_scaler.fit_transform(self.all_costs[:,np.newaxis])[:,0]
         self.scaled_uncers = self.all_uncers / self.cost_scaler.scale_
         self.gaussian_process.set_params(alpha=self.scaled_uncers**2)
+        if self.cost_has_noise:
+            if np.isnan(self.noise_level):
+                # Set noise_level to its default value, which is the variance of
+                # the training data. This will only happen on first iteration
+                # since self.noise_level is overwritten.
+                self.noise_level = self.cost_scaler.var_
```
Comment (charmasaur) [inline mloop/learners.py:1320]: I don't really have a sense of whether this is a sensible default. It seems to me as though it will massively overestimate the variance, but maybe the GP will quickly bring it back down to a reasonable value (I don't really know anything about GPs)?

Comment (zakv) [reply mloop/learners.py:1320]: Yes it is very likely an overestimate, probably too large by roughly a factor of the SNR squared. I chose that default because it was equivalent to the previous default value of 1 (after scaling). It might make sense to set it to 0.1 or 0.01 times the variance instead.

Ideally the GP will bring it down to a more reasonable value, but that's not always the case. I've sometimes seen the issue discussed in section 1.7.2.1. [here](https://scikit-learn.org/stable/modules/gaussian_process.html#gaussian-process) in which the fitting routine fits the data as essentially a flat signal with large noise, rather than as a curvy signal with moderate noise. I'm thinking about another PR to deal with that issue more directly, but setting the default value here to something smaller might help. In particular it might encourage the fitting routine to converge to the right local minimum of hyperparameter values. i.e. the one that explains the data as a curvy signal with a bit of noise.

Comment (charmasaur) [reply mloop/learners.py:1320]: Ah right, I hadn't thought about the default being equivalent to the old `1.0`, but yeah that makes sense. Thanks!
